source/slang/slang-ir-propagate-func-properties.cpp

```diff
@@ -238,34 +241,37 @@ class NoSideEffectFuncPropertyPropagationContext : public FuncPropertyPropagatio
         {
             for (auto inst : block->getChildren())
             {
-                // Is this inst known to not have global side effect/analyzable?
-                if (inst->mightHaveSideEffects())
+                switch (inst->getOp())
                 {
-                    switch (inst->getOp())
+                case kIROp_ifElse:
+                case kIROp_unconditionalBranch:
+                case kIROp_Switch:
+                case kIROp_Return:
+                case kIROp_loop:
+                case kIROp_Call:
+                case kIROp_Param:
+                case kIROp_Unreachable:
+                case kIROp_Store:
+                case kIROp_SwizzledStore:
+                    break;
```

Comment:
The duplication here with the logic above seems unfortunate. Is there any way to factor this out into a subroutine?

Reply:
Done.